test/parallel/test-repl-eval.js

```diff
@@ -0,0 +1,22 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const repl = require('repl');
+
+{
+  const options = {
+    eval: common.mustCall((cmd, context) => {
+      assert.strictEqual(cmd, 'foo\n');
+      assert.deepStrictEqual(context, {animal: 'Sterrance'});
+    })
+  };
+
+  const r = repl.start(options);
+  r.context = {animal: 'Sterrance'};
+
+  try {
+    r.write('foo\n');
+  } finally {
```

Comment:
Unless I'm mistaken, they'll only be ignored if there's a `catch` block. Since there's no `catch` block, the exceptions are propagated normally. (Tested by putting a `throw` call inside the `try` block.)